### PR TITLE
throwing now only randomly turns things if you are on harm intent

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

tin

## Why It's Good For The Game

i don't entirely hate the changes because making comfy nests out of bedsheets and pipes is fun but this is really annoying and letting people choose makes both sides happy far as i'm concerned.

## Changelog
:cl:
add: throwing things no longer makes them randomly turned as long as you aren't on harm intent
/:cl:

### CODE REVIEW

**Comment by GrayRachnid** [comment]
bruh you stole my pr i was jsut 2 minutes away from making a pr........

**Comment by GrayRachnid** [comment]
FUCK ME FOR TESTING MY CODE >:((((

**Comment by GrayRachnid** [comment]
Also you didn't apply this to drinks.dm and bottle.dm

**Comment by Linzolle** [comment]
good change

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -578,21 +578,22 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 			itempush = 0 //too light to push anything
 		return A.hitby(src, 0, itempush)
 
-/obj/item/throw_at(atom/target, range, speed, mob/thrower, spin=1, diagonals_first = 0, datum/callback/callback)
+/obj/item/throw_at(atom/target, range, speed, mob/thrower, spin=1, diagonals_first = 0, datum/callback/callback, messy_throw = TRUE)
 	thrownby = thrower
-	callback = CALLBACK(src, .proc/after_throw, callback) //replace their callback with our own
+	callback = CALLBACK(src, .proc/after_throw, callback, messy_throw) //replace their callback with our own
```
Comment (Ghommie) [inline code/game/objects/items.dm]:
```suggestion
	callback = CALLBACK(src, .proc/after_throw, callback, (spin && messy_throw)) //replace their callback with our own
```

Paper planes gliding smoothly only to end up instantly turned around has always been maddening me a whole lot.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -578,21 +578,22 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
...
-	M.Turn(rand(-170, 170))
-	transform = M
-	pixel_x = rand(-8, 8)
-	pixel_y = rand(-8, 8)
+	if(messy_throw)
+		var/matrix/M = matrix(transform)
+		M.Turn(rand(-170, 170))
+		transform = M
+		pixel_x = rand(-8, 8)
+		pixel_y = rand(-8, 8)
```
Comment (Ghommie) [inline code/game/objects/items.dm:596]: Hey, bonus points if you smooth this effect with animate().

**Review by Ghommie — COMMENTED** [review]
Honestly this is better off removed from mob's throwing entirely and left for pneumatic cannons, explosions and other things one could better define as messy. It makes little sense to restrict it harm intent only... moving on...

File: code/game/objects/items.dm (hunk 2 above)
Comment (silicons) [reply code/game/objects/items.dm:596]: uhh gimme a bit(tm)

**Comment by silicons** [comment]
@GrayRachnid lowkey thsi is a webeditor pr i did because people in ooc reminded me how much i hated this so

@Ghommie lmao the only reason i want this in at all is because it can be nice to make the "today i will sleep in a hallway in a pipe nest" thing, otherwise i'd want it gone from mob throwing entirely, it's just, annoying, it'd make sense on something like a 3d game but on byond with the limited pixel space and 2d stacking it just looks like shit and feels like shit in 90% of cases

**Comment by Ghommie** [comment]
Just use a table to carefully offset all the stuff you want to sleep under or onto, you junklord, bedsheets & co do shouldn't justify it.

File: code/game/objects/items.dm (hunk 2 above)
Comment (deathride58) [reply code/game/objects/items.dm:596]: Making this use animate() conflicts with the spin animation, which is why it doesn't already use animate()

Comment (Ghommie) [reply code/game/objects/items.dm:596]: Oh yea. crap.

**Comment by Linzolle** [comment]
literally why are you arguing the point of this PR is supposed to appease people who both like and don't like it

**Comment by Ghommie** [comment]
Add messy throwing back to explosions, repulse, sorium, black matter, grav gun and pneumatic cannons (can't remember what else) and it's ok to merge.

**Review by Ghommie — APPROVED** [review]
There are empty line endings changes you need to clean on carbon/carbon.dm, then I can merge it.

**Comment by AffectedArc07** [comment]
line ending changes aaaaaaaaaaaaaaaaaaaaaaaaaaaaa

**Pull request merged**